include/swift/Driver/Compilation.h

```diff
@@ -95,6 +96,11 @@ class Compilation {
   /// jobs itself.
   std::unique_ptr<const ToolChain> TheToolChain;
 
+  /// The OutputInfo, which the Compilation takes ownership of upon
+  /// construction, and which it may use to build subsequent batch
+  /// jobs itself.
+  std::unique_ptr<OutputInfo> TheOutputInfo;
```

Comment:
Ditto, ish. The Compilation isn't going to use this for anything *except* batching, and OutputInfo is a value-ish type anyway.

Reply:
@jrose-apple Same motive here; and here it would be more complex to re-manufacture the OutputInfo when needed later. Copy by-value?